lib/pdk/report.rb

```diff
@@ -94,22 +96,26 @@ def write_junit(target = self.class.default_target)
     # @param target [#write] an IO object that the report will be written to.
     #   Defaults to PDK::Report.default_target.
     def write_text(target = self.class.default_target)
-      # Open a File Object for IO if target is a string containing a filename or path
-      target = File.open(target, 'w') if target.is_a? String
       coverage_report = nil
+      report = []
 
       events.each do |_tool, tool_events|
         tool_events.each do |event|
           if event.rspec_puppet_coverage?
             coverage_report = event.to_text
           else
-            target.puts(event.to_text) unless event.pass?
+            report << event.to_text unless event.pass?
           end
         end
       end
-    ensure
```

Comment:
Shouldn't this ensure still be here?

Reply:
Nope, that ensure was only there to close the file, but that's handled by `PDK::Util::Filesystem.write_file` now.